### PR TITLE
Renamed all boot partition mentions to new firmware name convention

### PR DESCRIPTION
**Pull request by nathan-contino**
Addressing https://github.com/raspberrypi/documentation/issues/3089

### CODE REVIEW

**Comment by nathan-contino** [comment]
Unfortunately it looks like this include doesn't work quite right -- I can't seem to reference one include easily across multiple separate files in different directories. Oof. Open to suggestions. I don't think we necessarily *need* the note at every mention of the boot partition, but I suspect there are enough users who won't upgrade to Bookworm immediately that we should make this highly visible.

**Comment by aallan** [comment]
> Unfortunately it looks like this include doesn't work quite right…

Okay there are two things going on here that makes it not work. Firstly you need to include the file so that it's the relative path from the top level include, so you have a file called "foo.adoc" and it is including a file called "foo/bar.adoc" in a subdirectory called "foo", and you're including a file from a file in a different subdirectory "baz" then you need to do the relative path from the top level file, not the file in the subdirectory, because of the order things will get parsed in.

However, there seems to be a mismatch where the toolchain is checking for file existence from the subdirectory level, and not the top level. So if you get the path right so it'll pass the toolchain check, the file is actually in a different place one directory out from that. 

@lurch this is a toolchain problem, I've just done about an hour of testing and it's definitely doing something squirrely here.

The TL;DR is don't include a file from an included file, and then try and reference it from an entirely different chain. It tends to breaks things.

Sorry, but I really don't like this solution in any case. Scattering the same note throughout the text is messy. I think the best solution is to find the place where we talk about this file — and if there isn't one add one somewhere approrpiate — and then add a longer NOTE block there talking about the previous versions.

Yes, that's not perfect either. But we don't have the people to maintain documentation for previous versions.

**Comment by aallan** [comment]
So in summary,

1. rip the NOTE block out from everywhere you've put it
2. find the canonical reference for the `config.txt` file, and add a sightly bigger NOTE there.

File: documentation/asciidoc/computers/camera/libcamera_vid.adoc [hunk 1]
```diff
@@ -106,9 +106,11 @@ Using `libcamera-vid` to capture high framerate video (generally anything over 6
...
-4. It is advisable to set `force_turbo=1` in `/boot/config.txt` to ensure the CPU clock does not get throttled during the video capture. See https://www.raspberrypi.com/documentation/computers/config_txt.html#force_turbo[here] for further details.
+4. It is advisable to set `force_turbo=1` in `/boot/firmware/config.txt` to ensure the CPU clock does not get throttled during the video capture. See https://www.raspberrypi.com/documentation/computers/config_txt.html#force_turbo[here] for further details.
 5. Adjust the ISP output resolution with `--width 1280 --height 720` or something even lower to achieve your framerate target.
-6. On a Pi 4, you can overclock the GPU to improve performance by adding `gpu_freq=550` or higher in `/boot/config.txt`.  See https://www.raspberrypi.com/documentation/computers/config_txt.html#overclocking[here] for further details.
+6. On a Pi 4, you can overclock the GPU to improve performance by adding `gpu_freq=550` or higher in `/boot/firmware/config.txt`.  See https://www.raspberrypi.com/documentation/computers/config_txt.html#overclocking[here] for further details.
```
Comment (lurch) [inline documentation/asciidoc/computers/camera/libcamera_vid.adoc:111]: Eek. Not your fault @nathan-contino , but could you convert these https://www.raspberrypi.com/documentation/ absolute-links into the appropriate relative-link xrefs please?

File: documentation/asciidoc/computers/config_txt/note_moved_bookworm.adoc [hunk 2]
```diff
@@ -0,0 +1 @@
+NOTE: Prior to version 12 (_Bookworm_), Raspberry Pi OS stored the boot configuration in a partition named `/boot/`. Versions 12 renamed the boot partition to `/boot/firmware/`.
```
Comment (lurch) [inline documentation/asciidoc/computers/config_txt/note_moved_bookworm.adoc:1]: In Raspberry Pi OS, we don't really ever refer to the version-numbers (11, 12), we only mention the codenames (Bullseye, Bookworm).

File: documentation/asciidoc/computers/camera/libcamera_vid.adoc (hunk 1 above)
Comment (aallan) [reply documentation/asciidoc/computers/camera/libcamera_vid.adoc:111]: Please don't use "here" as a link word. The rule is that text should read properly, and be a real English sentence, without the link. Then insert the link into the sentence. There are a number of instances were "here" is used in the documentation (mostly due to third-party contributions before I started). But we're trying to strip them out.

Comment (aallan) [reply documentation/asciidoc/computers/camera/libcamera_vid.adoc:111]: So the xref here would be, `xref:config_txt.adoc#overclocking[link text]` (note the `.adoc` there!) as it's linking to another file inside the `computers/` section, if you were linking out to something in `accessories/` it'd be `xref:../accessories/foo.adoc[link text]`.

Comment (aallan) [reply documentation/asciidoc/computers/camera/libcamera_vid.adoc:111]: ...and this is why #3096 got opened. 😩

**Comment by aallan** [comment]
Opened #3103 about the toolchain bug. Nobody should look at that, it's just there so we remember it exists.

File: documentation/asciidoc/computers/camera/libcamera_vid.adoc (hunk 1 above)
Comment (lurch) [reply documentation/asciidoc/computers/camera/libcamera_vid.adoc:111]: > ... into the appropriate relative-link xrefs

I added a note to #3044

**Comment by nathan-contino** [comment]
Closing in favor of https://github.com/raspberrypi/documentation/pull/3105, which I don't have to rebase, and doesn't have this rather messy, broken commit in it 😄
